kernel/log.h

```diff
@@ -23,7 +23,22 @@
 #define LOG_H
 
 #include <time.h>
-#include <regex>
+#if defined(__GNUC__) && ( __GNUC__ == 4 && __GNUC_MINOR__ == 8)
+	#include <boost/xpressive/xpressive.hpp>
+	#define REGEX_TYPE boost::xpressive::sregex
+	#define REGEX_NS boost::xpressive
+	#define REGEX_COMPILE(param) boost::xpressive::sregex::compile(param, \
```

Comment:
Could we prefix those defines with `YS_` and add a comment here explaining why we need this? Also, is it only 4.8 that's broken or also gcc versions that predate 4.8? (not sure it would build on older gcc but it might be cleaner to check for <4.8 istead of =4.8 anyway.)

Reply:
Will add YS_ prefix and explanation, thing is that for version before 4.8 build will be broken anyway due to other c++11 missing features, I can however put <= just in case someone actually using it on ancient compilers.